sljit_src/sljitNativeMIPS_32.c

```diff
@@ -110,17 +139,31 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_fcopy(struct sljit_compiler *compi
 		else
 			FAIL_IF(push_inst(compiler, MFC1 | inst, DR(reg2)));
 
-		inst = FS(freg) | (1 << 11);
+#if defined(SLJIT_MIPS_REV) && SLJIT_MIPS_REV >= 2
+		if (cpu_feature_list & CPU_FEATURE_FR) {
+			mthc = MTHC1;
+			mfhc = MFHC1;
+			inst = FS(freg);
+		} else
+#endif /* SLJIT_MIPS_REV >= 2 */
+			inst = FS(freg) | (1 << 11);
```

Comment:
You don't need else here, just keep the original code. One less branch instruction as well. `inst ^= (1 << 11)` can be used as well.

Reply:
While I agree the else is a little off, there is IMHO no way to get rid of the branch (at least for `SLJIT_MIPS_REV >= 2` which I hope is the most common path.

Will do the double assignment of `inst` to remove the else as instructed, with the hope that the compiler will optimize it somehow.

Reply:
The point is, if the compiler does not optimize it, it is the better. Setting a value is cheap, and you don't need a branch at the end.